### PR TITLE
typo correction for bio_005, comments for cli_021

### PR DESCRIPTION
**Pull request by Taufiq06**

### CODE REVIEW

File: cli_021_snow_cover/contents/src/__init__.py [hunk 1]
```diff
@@ -14,328 +14,496 @@
...
 SOURCE_URL = 'https://n5eil01u.ecs.nsidc.org/MOST/MOD10CM.006/{date}'
+
+# subdataset to be converted to tif
+# should be of the format 'HDF4_EOS:EOS_GRID:"filename.hdf":variable'
 SDS_NAME = 'HDF4_EOS:EOS_GRID:"{fname}":MOD_CMG_Snow_5km:Snow_Cover_Monthly_CMG'
+
+# filename format for GEE
 FILENAME = 'cli_021_{date}'
+
+# nodata value for netcdf
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py]:
```suggestion
# nodata value for hdf
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 2]
```diff
@@ -14,328 +14,496 @@
...
+            # read the opened url
+            content = response.read()
+            # use BeautifulSoup to read the content as a nested data structure
+            soup = BeautifulSoup(content, 'html.parser')
+
+            hdfs = []
+            # extract all the <a> tags within the content, the goal is to find id of all the hdf files
+            for a in soup.find_all('a'):
+                str_a = str(a)
+                # if an <a> tag contain the word 'hdf'
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:331]:
```suggestion
                # if one of the links available to download contains the word 'hdf'
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 3]
```diff
@@ -14,328 +14,496 @@
...
+        try:
+            # open the url
+            response = urllib.request.urlopen(url)
+            # read the opened url
+            content = response.read()
+            # use BeautifulSoup to read the content as a nested data structure
+            soup = BeautifulSoup(content, 'html.parser')
+
+            hdfs = []
+            # extract all the <a> tags within the content, the goal is to find id of all the hdf files
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:328]:
```suggestion
            # Extract all the <a> tags within the html content. The <a> tags are used to mark links, so 
            # we will be able to find the files available for download marked with these tags.
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 4]
```diff
@@ -14,328 +14,496 @@
...
+        urllib.request.HTTPBasicAuthHandler(password_manager),
+        urllib.request.HTTPCookieProcessor(cookie_jar))
+    # install an OpenerDirector instance as the default global opener
+    urllib.request.install_opener(opener)
+
+    # make an empty list to store names of the files we downloaded
+    files = []
+    # go through each input date
+    for date in new_dates:
+        # get the url to download the file from the source for the given date
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:313]:
```suggestion
        # get the url where data for the given date is stored at the source
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 5]
```diff
@@ -14,328 +14,496 @@
...
+            # use BeautifulSoup to read the content as a nested data structure
+            soup = BeautifulSoup(content, 'html.parser')
+
+            hdfs = []
+            # extract all the <a> tags within the content, the goal is to find id of all the hdf files
+            for a in soup.find_all('a'):
+                str_a = str(a)
+                # if an <a> tag contain the word 'hdf'
+                if 'hdf' in str_a:
+                    # find the index of '.hdf'
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:333]:
```suggestion
                    # get the name of the hdf file
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 6]
```diff
@@ -14,328 +14,496 @@
...
+                    ext = str_a.index('.hdf')
+                    # slice the id of the .hdf files using the index
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:335]:
```suggestion
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 7]
```diff
@@ -14,328 +14,496 @@
...
+                    hdf = str_a[9:ext+4]
+                    # store all the hdf ids to a list
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:337]:
```suggestion
                    # add the hdf file name to the list of files to download
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 8]
```diff
@@ -14,328 +14,496 @@
...
+                # try to download the data
+                urllib.request.urlretrieve(url, f)
+                # if successful, add the file to the list of files we have downloaded
+                files.append(f)
+                # if successful, log that the file was downloaded successfully
+                logging.info('Successfully retrieved {}'.format(f))
+
+            except Exception as e:
+                # if unsuccessful, log that the file was not downloaded
+                # (could be because we are attempting to download a file that is not available yet)
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:357]: We already checked that this file was on the website, so we know the file is available--something else would have caused the error
```suggestion
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 9]
```diff
@@ -14,328 +14,496 @@
...
+            try:
+                # try to download the data
+                urllib.request.urlretrieve(url, f)
+                # if successful, add the file to the list of files we have downloaded
+                files.append(f)
+                # if successful, log that the file was downloaded successfully
+                logging.info('Successfully retrieved {}'.format(f))
+
+            except Exception as e:
+                # if unsuccessful, log that the file was not downloaded
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:356]:
```suggestion
                # if unsuccessful, log an error that the file was not downloaded
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 10]
```diff
@@ -14,328 +14,496 @@
...
-            for tif in tifs:
-                os.remove(tif)
-            for f in files:
-                os.remove(f)
-
-        return assets
-    return []
+    '''
+    fetch, process, upload, and clean new data
+    INPUT   existing_dates: list of dates we already have in GEE (list of strings)
```
Comment (amsnyder) [inline cli_021_snow_cover/contents/src/__init__.py:371]:
```suggestion
    INPUT   existing_dates: list of dates we already have in GEE, in the format of the DATE_FORMAT variable (list of strings)
```

File: cli_021_snow_cover/contents/src/__init__.py [hunk 11]
```diff
@@ -14,328 +14,496 @@
...
+
+# subdataset to be converted to tif
+# should be of the format 'HDF4_EOS:EOS_GRID:"filename.hdf":variable'
 SDS_NAME = 'HDF4_EOS:EOS_GRID:"{fname}":MOD_CMG_Snow_5km:Snow_Cover_Monthly_CMG'
+
+# filename format for GEE
 FILENAME = 'cli_021_{date}'
+
+# nodata value for netcdf
+# this netcdf has a nodata value of -5
```
[the remark this answers is not in the document]
Comment (Taufiq06) [reply cli_021_snow_cover/contents/src/__init__.py:28]: So, the nodata value is not -5 for this dataset?

**Pull request merged**